test/hmac4/test_hmac4.py

```diff
@@ -30,7 +30,7 @@ def request_from_text(text):
 
     """
     lines = text.splitlines()
-    match = re.search("^([a-z]+) (.*) (http/[0-9]\.[0-9])$", lines[0], re.I)
+    match = re.search(r"^([a-z]+) (.*) (http/[0-9]\.[0-9])$", lines[0], re.I)
```

Comment:
using raw string here to fix a warning `DeprecationWarning: invalid escape sequence \.`
see https://stackoverflow.com/questions/52335970/how-to-fix-string-deprecationwarning-invalid-escape-sequence-in-python